niworkflows/workflows/gradunwarp.py

```diff
@@ -64,6 +64,29 @@ def init_gradunwarp_wf(
         iterfield=["reference", "warp1"]
     )
 
+    def _warp_fsl2itk(in_warp):
+        import os
+        import nitransforms.io
+        from nipype.utils.filemanip import fname_presuffix
+        fsl_warp = nitransforms.io.fsl.FSLDisplacementsField.from_filename(in_warp)
+        itk_warp_data = fsl_warp.get_fdata().reshape(fsl_warp.shape[:3]+(1,3))
+        itk_warp_data[...,(0,1)] *= -1
+        itk_warp = fsl_warp.__class__(itk_warp_data, fsl_warp.affine)
+        itk_warp.header.set_intent("vector")
+        out_fname = fname_presuffix(in_warp, suffix="_itk", newpath=os.getcwd())
+        itk_warp.to_filename(out_fname)
```

Comment:
nitransforms should do all the manipulations for you

Reply:
I might have missed something, but from what I understood what needs to be provided to Ants ApplyTransform are ITK warpfields. In nitransforms I cannot see any function to output non-linear warp in ITK format from nitransforms object https://nitransforms.readthedocs.io/en/latest/_api/io.html#nitransforms.io.itk.ITKDisplacementsField, it seems to only deal with inputs. Thus, my attempt to invert that input step.

Reply:
Yeah, it looks like you are totally right (and I did the same here a while ago - https://github.com/nipreps/sdcflows/blob/375dd7a024abcd5d4d2b282241a037d76f9719e6/sdcflows/transform.py#L286-L292)

Would you like to become a proud contributor to nitransforms? :D

If not (or either way, actually), may I ask you to file an issue requesting this to be implemented there?